### PR TITLE
Kernel Interrupt api

### PR DESCRIPTION
**Pull request by colombod**

### CODE REVIEW

File: Microsoft.DotNet.Interactive/Commands/InterruptKernel.cs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 namespace Microsoft.DotNet.Interactive.Commands
 {
-    public class InterruptExecution : KernelCommandBase
+    public class InterruptKernel : KernelCommandBase
```
Comment (jonsequitur) [inline Microsoft.DotNet.Interactive/Commands/InterruptKernel.cs]: The thing being interrupted isn't the kernel. It's an operation or an invocation or something.

File: Microsoft.DotNet.Interactive.Jupyter.Tests/InterruptRequestHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DotNet.Interactive.Jupyter.Tests
+{
+    public class InterruptRequestHandlerTests
+    {
+
+    }
```
Comment (jonsequitur) [inline Microsoft.DotNet.Interactive.Jupyter.Tests/InterruptRequestHandlerTests.cs]: So tested!

File: Microsoft.DotNet.Interactive.Jupyter.Tests/InterruptRequestHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            var kernel = new CSharpKernel();
+            var handler = new InterruptRequestHandler(kernel);
+            var request = Message.Create(new DisplayData(), null);
+            Func<Task> messageHandling = () => handler.Handle(new JupyterRequestContext(_serverChannel, _ioPubChannel, request, _kernelStatus));
+            messageHandling.Should().ThrowExactly<InvalidOperationException>();
+        }
+
+        [Fact]
+        public async Task handles_InterruptRequest()
```
Comment (akshita31) [inline Microsoft.DotNet.Interactive.Jupyter.Tests/InterruptRequestHandlerTests.cs]: Should we merge this one and the one below into one test as this one doesnt have any assertion ?

Comment (colombod) [reply Microsoft.DotNet.Interactive.Jupyter.Tests/InterruptRequestHandlerTests.cs]: it would throw if not using the right message

Comment (akshita31) [reply Microsoft.DotNet.Interactive.Jupyter.Tests/InterruptRequestHandlerTests.cs]: Ok. Thanks!

**Pull request merged**